examples/experimental/dagster-airlift/dagster_airlift/core/sensor.py

```diff
@@ -196,11 +199,7 @@ def materializations_and_requests_from_batch_iter(
             for asset_key in asset_keys:
                 spec = repo_def.assets_defs_by_key[asset_key].get_asset_spec(asset_key)
                 partition_key = (
-                    get_partition_key_from_task_instance(
-                        partitions_def=spec.partitions_def,
-                        task_instance=task_run,
-                        asset_key=asset_key,
-                    )
+                    partition_resolver(spec.partitions_def, task_run, asset_key)
                     if spec.partitions_def
                     else None
                 )
```

Comment:
<!--__GRAPHITE_HTML_TAG_START__--><i class='graphite__hidden'>[Re: lines +202 to +214]<br/><br/></i>

<!--__GRAPHITE_HTML_TAG_END__-->Hmmm I think we need to consider pluggability holistically. We don't want to end up with a huge number of callbacks I think. This is a good opportunity to allow the user to customize metadata and all sorts of stuff.<!--__GRAPHITE_HTML_TAG_START__--><span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24807?utm_source=unchanged-line-comment">Graphite</a>.</span>

<!--__GRAPHITE_HTML_TAG_END__-->